src/gui/optionsdialog.cpp

```diff
@@ -489,6 +492,36 @@ void OptionsDialog::initializeLanguageCombo()
     }
 }
 
+
+
+void OptionsDialog::initializeThemeCombo()
+{
+    m_ui->comboTheme->addItem(tr("Default"));
+    if (UIThemeManager::instance()->isCustomAvailable())
+        m_ui->comboTheme->addItem(tr("Custom"));
+    m_ui->comboTheme->insertSeparator(m_ui->comboTheme->count());
+    m_ui->comboTheme->addItem(tr("Select..."));
+    m_ui->comboTheme->setCurrentIndex(UIThemeManager::instance()->theme() == UIThemeManager::Theme::Default ? 0 : 1);
+
+    connect(m_ui->comboTheme, qOverload<int>(&QComboBox::currentIndexChanged), [&](int index)
+    {
+        if (index == (m_ui->comboTheme->count() - 1)) {
+            m_qssFilePath = QFileDialog::getOpenFileName(this, tr("Select Qss File"), "", tr("Style File (*.qss)"));
+            if (!m_qssFilePath.isEmpty()) {
+                m_ui->comboTheme->blockSignals(true);
+                if (m_ui->comboTheme->count() == 3)
+                    m_ui->comboTheme->insertItem(1, tr("Custom"));
+                m_ui->comboTheme->setCurrentIndex(1);
+                m_ui->comboTheme->blockSignals(false);
+            }
+            else {
+                // don't leave `Select...` as current text
+                m_ui->comboTheme->setCurrentIndex(UIThemeManager::instance()->theme() == UIThemeManager::Theme::Default ? 0 : 1);
```

Comment:
Ok, it is reasonable.
Also you can move block/unblock signals outside if/else to prevent unnecessary signal here.